### DISCUSSION TITLE
import database functionality of webui.db

### DISCUSSION THREAD

**Discussion by arjunkrishna** [open-webui/open-webui#1585] [Ideas]
it would be a nice to have feature to have functionality to import webui.db under admin panel>database

i run open-webui within docker desktop on windows 11 machine with wsl enabled. was looking for an easy way to restore webui.db if needed in the future.

**Comment by gruckion** [comment]
More information please what is webui.db?

**Comment by arjunkrishna** [comment]
under admin settings, we can download database, but we at present cannot upload/restore a previous copy of the database.  sample screenshot 
![image](https://github.com/open-webui/open-webui/assets/5271912/7b176644-77f4-4d52-ba97-fccf14250097)
****

**Reply by nnnarvaez** [reply]
cp ./oldatabase/webui.db ./newfolder/

**Comment by x-limitless-x** [comment]
I hope this can get worked on, it would be very helpful to be able to import the database from the webui.
![image](https://github.com/open-webui/open-webui/assets/17127926/bd8b08ea-2004-464f-8ebf-2878560084db)

**Comment by RickyEsclapon** [comment]
I'm moving from an old macbook to a new one - seems weird that I would have no way of importing my previous db. There's probably a way to do this without the UI explicitly supporting this, or is that not the case?

**Comment by arjunkrishna** [comment]
https://github.com/open-webui/open-webui/discussions/4157

**Comment by mnawaz6935** [comment]
To attach the database file stored at `D:\OpenWebUI\webui.db` in your Docker command, you need to map the local file path to the container's file system. Here's the revised command:

```bash
docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v D:\OpenWebUI\webui.db:/app/backend/data/webui.db -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

### Explanation:
- `-v D:\OpenWebUI\webui.db:/app/backend/data/webui.db`: This maps the local `webui.db` file from `D:\OpenWebUI\` to the container's `/app/backend/data/webui.db` path.
- The rest of the command remains the same, ensuring the container runs in detached mode (`-d`), maps port `3000` to `8080`, and uses the `open-webui` volume for other data.

Make sure the file path is correct and accessible from your Docker environment. If you're using Docker on Windows, ensure the drive is shared with Docker Desktop.

**Comment by bstrdsmkr** [comment]
FYI, this only works when you have direct access to the underlying storage, so the feature request here is still valid.

For example, if you have a helm chart deployment, the team that manages _**application**_ (open-webui) might not be the same team that manages _**deployment**_ of the application (a DevOps team), nor the team that manages the Kubernetes instance.

Given that, it's reasonable to see a situation where the team administering the application doesn't have access to upload the database to the pod's storage and would have to go through another team to get it done.
